src/services/services.ts

```diff
@@ -6092,13 +6195,32 @@ namespace ts {
                 findInStrings: boolean,
                 findInComments: boolean,
                 result: ReferencedSymbol[],
-                symbolToIndex: number[]): void {
+                symbolToIndex: number[],
+                indexToSymbol: {[index: number]: Symbol}): void {
 
                 const sourceFile = container.getSourceFile();
 
                 const start = findInComments ? container.getFullStart() : container.getStart();
                 const possiblePositions = getPossibleSymbolReferencePositions(sourceFile, searchText, start, container.getEnd());
 
+                // If we are just looking for implementations and this is a property access expression, we need to get the
+                // symbol of the local type of the symbol the property is being accessed on. This is because our search
+                // symbol may have a different parent symbol if the local type's symbol does not declare the property
+                // being accessed (i.e. it is declared in some parent class or interface)
+                let parentSymbols: SymbolInheritanceState[] = undefined;
+
+                if (implementations && isRightSideOfPropertyAccess(searchLocation)) {

```

Comment:
`isTypeRelatedTo` has a similar implementation for caching that you may find helpful.